src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = String(this[0]);
+
+    if (this[0] === null || this[0] === undefined) {
+      result = '';
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += separator + '';
+      } else {
+        result += separator + String(this[i]);
+      }
```

Comment:
This logic can be simplified, just don't add anything if element is null or undefined. (adding empty string and not adding anything leads to same result)